tools/packaging/catalog/catalog-remote.js

```diff
@@ -435,6 +439,41 @@ Object.assign(Table.prototype, {
     self._deleteQuery = "DELETE FROM " + self.name + " WHERE _id=?";
   },
 
+  // Compress a JSON string for storage
+  _compressString: function(str) {
+    var self = this;
+    if (!self._compressContent) {
+      return str;
+    }
+    
+    try {
+      var compressed = zlib.gzipSync(Buffer.from(str, 'utf8'));
+      // Mark as compressed with a prefix to distinguish from uncompressed data
+      return 'GZIP:' + compressed.toString('base64');
+    } catch (e) {
+      // If compression fails, fall back to uncompressed
+      Console.debug('Catalog compression failed, using uncompressed data:', e.message);
+      return str;
+    }
+  },
+
+  // Decompress a string that might be compressed
+  _decompressString: function(str) {
+    if (typeof str === 'string' && str.startsWith('GZIP:')) {
```

Comment:
To improve readability, would you mind inverting this condition? It will remove one indentation level, as it did in. `_compressString`

``` js
if(!(typeof str === 'string' && str.startsWith('GZIP:'))) return str
try {
....
} catch (e) { 
... 
}
```